File: nunchuk-widget/src/main/java/com/nunchuk/android/widget/NCFontTextView.java
package com.nunchuk.android.widget;

import android.content.Context;
import android.content.res.TypedArray;
import android.util.AttributeSet;

import androidx.appcompat.widget.AppCompatTextView;

public class NCFontTextView extends AppCompatTextView {

    private String fontName;
    private int fontStyle;

    public NCFontTextView(final Context context) {
        super(context);
        setTypeface(FontStyle.getCurrentTypeface(getContext(), fontStyle, fontName));
    }

    public NCFontTextView(final Context context, final AttributeSet attrs) {
        super(context, attrs);
        getFontNameFromAttr(attrs);
        if (!isInEditMode()) {  // check isInEditMode to use FontTextView in xml editor without error
            setTypeface(FontStyle.getCurrentTypeface(getContext(), fontStyle, fontName));
        }
    }

    public NCFontTextView(final Context context, final AttributeSet attrs, final int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        getFontNameFromAttr(attrs);
        setTypeface(FontStyle.getCurrentTypeface(getContext(), fontStyle, fontName));
    }

    private void getFontNameFromAttr(final AttributeSet attrs) {
        final TypedArray typedArray = getContext().obtainStyledAttributes(attrs, R.styleable.FontTextView);
        if (typedArray != null) {
            fontName = typedArray.getString(R.styleable.FontTextView_fontTextName);
            fontStyle = typedArray.getInt(R.styleable.FontTextView_fontTextStyle, 0);
            typedArray.recycle();
        }
    }

}

File: nunchuk-widget/src/main/java/com/nunchuk/android/widget/NCFontTextView.kt
package com.nunchuk.android.widget

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatTextView
import com.nunchuk.android.widget.util.FontInitializer

class NCFontTextView : AppCompatTextView {

    private val initializer: FontInitializer by lazy { FontInitializer(context) }

    constructor(context: Context) : super(context) {
        initializer.initTypeface(this)
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        initializer.initTypeface(this, attrs)
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        initializer.initTypeface(this, attrs)
    }
}
